qiskit/providers/ibmq/api/session.py

```diff
@@ -259,3 +268,74 @@ def _modify_chained_exception_messages(self, exc: BaseException) -> None:
+
+        Note:
+            The string ``...`` is used to denote information that has been filtered out
+            from the request, within the url and request data. Currently, the backend name
+            is filtered out from endpoint URLs, using a regex to capture the name, and from
+            the data sent to the server when submitting a job.
+
+            The request data is only logged for the following URLs, since they contain useful
+            information: ``/Jobs`` (POST), ``/Jobs/status`` (GET),
+            and ``/devices/<device_name>/properties`` (GET).
+
+        Args:
+            url: URL for the new request.
+            method: Method for the new request (e.g. ``POST``)
+            request_data:Additional arguments for the request.
+
+        Raises:
+            Exception: If there was an error logging the request information.
+        """
+        # Replace the device name in the URL with `...` if it matches, otherwise leave it as is.
+        filtered_url = re.sub(RE_DEVICES_ENDPOINT, '\\1...\\3', url)
+
+        if self._is_worth_logging(filtered_url):
+            try:
+                method = method.upper()
+
+                if filtered_url in ('/Jobs/status', '/devices/.../properties', '/Jobs'):
+                    if filtered_url == '/Jobs' and 'json' in request_data:
+                        # Replace backend name in job submission request.
+                        request_data['json']['backend']['name'] = '...'
+                    logger.debug('Endpoint: %s. Method: %s. Request Data: %s.',
+                                 filtered_url, method, request_data)
+                else:
+                    logger.debug('Endpoint: %s. Method: %s.', filtered_url, method)
+            except Exception as ex:  # pylint: disable=broad-except
+                # Catch general exception so as not to disturb the program if filtering fails.
+                logger.info('Filtering failed when logging request information: %s', str(ex))
+
+    def _is_worth_logging(self, endpoint_url: str) -> bool:
+        """Returns whether the endpoint URL should be logged.
+
+        The checks in place help filter out endpoint URL logs that would add noise
+        and no helpful information.
+
+        Note:
+            The following endpoint URLs are not logged: ``/devices/v/1`` and
+            ``/devices/<device_name>/queue/status``. Likewise, the endpoint URLs that start
+            with ``/users`` or ``/version``, or contain 'objectstorage', are not logged.
+
+        Args:
+            endpoint_url: The endpoint URL that will be logged.
+
+        Returns:
+            Whether the endpoint URL should be logged.
+        """
+        if endpoint_url in ('/devices/.../queue/status', '/devices/v/1'):
+            return False
+        if endpoint_url.startswith(('/users', '/version')):
+            return False
+        if 'objectstorage' in endpoint_url:
+            return False
```

Comment:
It was mentioned that the object storage endpoint url should not be logged. It's a bit more difficult to filter since the `url` is not an endpoint but instead an absolute url. I did notice that `objectstorage` is contained in the base url and filtered for it, since I couldn't think of another way to filter for it...